File: client/src/types/JsonModels/Analytics/PersonalRecordSummaryModel.ts
export interface PersonalRecordSummaryModel
{
	exerciseId: number;
	exerciseName: string;
	primaryMuscleGroupId: number;
	primaryMuscleGroupName: string;
	bestWeightKg?: number;
	bestReps?: number;
	bestEstimatedOneRepMax?: number;
	bestVolumeKg?: number;
	lastTrainedOn: string;
}

File: server/FitMate.Core/JsonModels/Analytics/PersonalRecordSummaryModel.cs
namespace FitMate.Core.JsonModels.Analytics;

public class PersonalRecordSummaryModel
{
    public long ExerciseId { get; set; }
    public string ExerciseName { get; set; } = string.Empty;
    public long PrimaryMuscleGroupId { get; set; }
    public string PrimaryMuscleGroupName { get; set; } = string.Empty;
    public decimal? BestWeightKg { get; set; }
    public int? BestReps { get; set; }
    public decimal? BestEstimatedOneRepMax { get; set; }
    public decimal? BestVolumeKg { get; set; }
    public DateTime LastTrainedOn { get; set; }
}
